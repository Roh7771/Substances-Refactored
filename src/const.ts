export const SUBSTANCE_TO_SHOW_AMOUNT = 20;

export const EMPTY_FIELD_VALUE = '-';

export enum ErrorStatus {
  OK,
  DUPLICATE_CAS_NUMBER,
  LOADING_FAILED,
  WRONG_LOGIN_DATA
}

export enum AuthorizationStatus {
  AUTH,
  NO_AUTH
}

export enum ModalWindowStatus {
  EDIT = 'EDIT',
  CREATE = 'CREATE',
  DELETE = 'DELETE',
  NONE = 'NONE'
}
