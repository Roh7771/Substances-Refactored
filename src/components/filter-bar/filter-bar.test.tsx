import * as React from 'react';
import * as renderer from 'react-test-renderer';
import FilterBar from './filter-bar';
import { ErrorStatus } from '../../const';

const noop = () => null;

it('Filter bar should render correctly', () => {
  const tree = renderer.create(
    <FilterBar
      dispatch={noop}
      errorStatus={ErrorStatus.OK}
      locationCollection={
        new Map([
          [111, new Set(['first place', 'second place'])],
          [222, new Set(['first place', 'second place'])],
        ])
      }
      chosenLocations={[111]}
    />,
  );
  expect(tree).toMatchSnapshot();
});

it('Filter bar should not render when there is an error', () => {
  const tree = renderer.create(
    <FilterBar
      dispatch={noop}
      errorStatus={ErrorStatus.LOADING_FAILED}
      locationCollection={
        new Map([
          [111, new Set(['first place', 'second place'])],
          [222, new Set(['first place', 'second place'])],
        ])
      }
      chosenLocations={[111]}
    />,
  );
  expect(tree).toMatchSnapshot();
});
