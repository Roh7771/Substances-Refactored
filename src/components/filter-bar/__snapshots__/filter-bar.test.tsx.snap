// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter bar should not render when there is an error 1`] = `
<div
  className="filter-bar"
  style={
    Object {
      "display": "none",
    }
  }
>
  <h2
    className="filter-bar__text"
  >
    Фильтр по лабораториям:
  </h2>
  <div
    className="form-check form-check-inline"
  >
    <input
      checked={true}
      className="form-check-input"
      id="lab-111"
      onChange={[Function]}
      type="checkbox"
      value={111}
    />
    <label
      className="form-check-label"
      htmlFor="lab-111"
    >
      111
    </label>
  </div>
  <div
    className="form-check form-check-inline"
  >
    <input
      checked={false}
      className="form-check-input"
      id="lab-222"
      onChange={[Function]}
      type="checkbox"
      value={222}
    />
    <label
      className="form-check-label"
      htmlFor="lab-222"
    >
      222
    </label>
  </div>
</div>
`;

exports[`Filter bar should render correctly 1`] = `
<div
  className="filter-bar"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <h2
    className="filter-bar__text"
  >
    Фильтр по лабораториям:
  </h2>
  <div
    className="form-check form-check-inline"
  >
    <input
      checked={true}
      className="form-check-input"
      id="lab-111"
      onChange={[Function]}
      type="checkbox"
      value={111}
    />
    <label
      className="form-check-label"
      htmlFor="lab-111"
    >
      111
    </label>
  </div>
  <div
    className="form-check form-check-inline"
  >
    <input
      checked={false}
      className="form-check-input"
      id="lab-222"
      onChange={[Function]}
      type="checkbox"
      value={222}
    />
    <label
      className="form-check-label"
      htmlFor="lab-222"
    >
      222
    </label>
  </div>
</div>
`;
